### PR TITLE
feat(astro): Support `auth().redirectToSignIn()` at page level

### PR DESCRIPTION
**Pull request by panteliselef**
## Description

Support `Astro.locals.auth().redirectToSignIn()`
This allows for redirectingToSignIn at the page level and provides parity with Nextjs.

Note: We are not exposing `auth().protect()` because DX is taking a hit, because we cannot just throw a response in astro, we need to return it. And we are not including protect at the middleware level so we don't confuse people that would expect it to work at page level.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/astro/src/server/clerk-middleware.ts [hunk 1]
```diff
@@ -218,14 +224,38 @@ Check if signInUrl is missing from your configuration or if it is not an absolut
...
+
+      return createRedirect({
+        redirectAdapter,
+        devBrowserToken: devBrowserToken,
+        baseUrl: clerkUrl.toString(),
+        publishableKey: getSafeEnv(context).pk!,
+        signInUrl: requestState.signInUrl,
+        signUpUrl: requestState.signUpUrl,
+      }).redirectToSignIn({
+        returnBackUrl: opts.returnBackUrl === null ? '' : opts.returnBackUrl || clerkUrl.toString(),
```
Comment (wobsoriano) [inline packages/astro/src/server/clerk-middleware.ts:251]: Does the `redirectToSignUp` function commonly used?

**Review by wobsoriano — APPROVED** [review]
Looks straightforward, just left a question 🚀

File: packages/astro/src/server/clerk-middleware.ts (hunk 1 above)
Comment (panteliselef) [reply packages/astro/src/server/clerk-middleware.ts:251]: We've build it, but no public API consumes it. and this implementation aims to mirror the API from @clerk/nextjs.

Comment (wobsoriano) [reply packages/astro/src/server/clerk-middleware.ts:251]: Thanks for answering!

**Pull request merged**